net: ip: net_pkt: Fix net_pkt_pull data buffer corruption

a8961713 opened the pull request:
Fixes #22307

tbursztyka reviewed (commented):
looks like you inverted the commit messages and the actual patches

a8961713 commented:
> looks like you inverted the commit messages and the actual patches

You are correct. I'll resubmit.

jukkar commented on subsys/net/ip/net_pkt.c:
```diff
@@ -1848,7 +1848,7 @@ int net_pkt_pull(struct net_pkt *pkt, size_t length)
 	net_pkt_cursor_backup(pkt, &backup);
```
No need to duplicate the "net" prefix so many times in commit subject, for the first commit you can just say `net: pkt: Fixed packet buffer corruption in net_pkt_pull()`

jukkar commented on tests/net/net_pkt/src/main.c:
```diff
@@ -629,6 +629,49 @@ void test_net_pkt_copy(void)
 		     "Pkt not properly unreferenced");
```
for the second commit, please change the subject to `tests: net: pkt: Added a test case for net_pkt_pull()`

jukkar reviewed (changes requested):
Looks good, some minor nits.

pfalcon commented on tests/net/net_pkt/src/main.c:
```diff
@@ -629,6 +629,49 @@ void test_net_pkt_copy(void)
+	zassert_mem_equal(pkt_data_readback,
+			  &pkt_data[PULL_AMOUNT],
+			  PULL_TEST_PKT_DATA_SIZE - PULL_AMOUNT,
+			  "Pakcet data changed");
```
Typo: "Pakcet".

a8961713 replied:
fixed, resubmitted.

pfalcon reviewed (changes requested):
Typo.

a8961713 committed:
net: pkt: Fixed packet buffer corruption in net_pkt_pull()

Changed size related variables type to size_t to handle
large packet buffers correctly

Signed-off-by: David D <a8961713@gmail.com>

a8961713 committed:
test: net: pkt: Added a test case for net_pkt_pull()

Added a test case for net_pkt_pull() with large packet buffers

Signed-off-by: David D <a8961713@gmail.com>
